novawallet/Modules/Staking/StakingSetupProxy/Validation/ProxyDataValidatorFactory.swift

```diff
@@ -0,0 +1,163 @@
+    }
+
+    func hasSufficientBalance(
+        available: BigUInt?,
+        deposit: BigUInt?,
+        fee: BigUInt?,
+        asset: AssetBalanceDisplayInfo,
+        locale: Locale
+    ) -> DataValidating {
+        ErrorConditionViolation(onError: { [weak self] in
+            guard
+                let view = self?.view,
+                let viewModelFactory = self?.balanceViewModelFactoryFacade else {
+                return
+            }
+            let balanceDecimal = available?.decimal(assetInfo: asset)
+            let depositDecimal = deposit?.decimal(assetInfo: asset)
+
+            let balanceModel = viewModelFactory.amountFromValue(
+                targetAssetInfo: asset,
+                value: balanceDecimal ?? 0
+            ).value(for: locale)
+            let depositModel = viewModelFactory.amountFromValue(
+                targetAssetInfo: asset,
+                value: depositDecimal ?? 0
+            ).value(for: locale)
+
+            self?.presentable.presentNotEnoughBalanceForDeposit(
+                from: view,
+                deposit: depositModel,
+                balance: balanceModel,
+                locale: locale
+            )
+        }, preservesCondition: {
+            guard let deposit = deposit,
+                  let fee = fee,
+                  let available = available else {
+                return false
+            }
+            return available >= deposit + fee
+        })
+    }
+
+    func validAddress(
+        _ address: String,
+        chain: ChainModel,
+        locale: Locale
+    ) -> DataValidating {
+        ErrorConditionViolation(onError: { [weak self] in
+            guard let view = self?.view else {
+                return
+            }
+            self?.presentable.presentNotValidAddress(
+                from: view,
+                networkName: chain.name,
+                locale: locale
+            )
+        }, preservesCondition: {
+            let accountId = try? address.toAccountId(using: chain.chainFormat)
+            return accountId != nil
+        })
+    }
+
+    func notReachedMaximimProxyCount(
+        _ proxyCount: Int?,
+        limit: Int?,
+        chain: ChainModel,
+        locale: Locale
+    ) -> DataValidating {
+        ErrorConditionViolation(onError: { [weak self] in
+            guard let view = self?.view else {
+                return
+            }
+
+            let limitNumber = NSNumber(value: limit ?? 0)
+            let formatter = NumberFormatter.quantity
+            self?.presentable.presentMaximumProxyCount(
+                from: view,
+                limit: formatter.string(from: limitNumber) ?? "",
+                networkName: chain.name,
+                locale: locale
+            )
+        }, preservesCondition: {
+            guard let proxyCount = proxyCount,
+                  let limit = limit else {
+                return false
+            }
+            return proxyCount < limit
```

Comment:
I would assume that proxyCount is new number of proxies. In this case we must have ```proxyCount <= limit```